chain/client/src/chunk_distribution_network.rs

```diff
@@ -441,14 +441,14 @@ mod tests {
             futures::future::ready(Ok(chunk))
         }
 
-        fn publish_chunk(
+        async fn publish_chunk(
             &mut self,
             chunk: &PartialEncodedChunk,
-        ) -> impl Future<Output = Result<Self::Response, Self::Error>> {
+        ) -> Result<Self::Response, Self::Error> {
             let prev_hash = *chunk.prev_block();
             let shard_id = chunk.shard_id();
             self.known_chunks.insert((prev_hash, shard_id), chunk.clone());
-            futures::future::ready(Ok(()))
+            Ok(())
         }
     }
```

Comment:
This is functionally different. The first option executes the `insert` before the future is first polled, the 2nd option will only do that after the first poll of the future. This appears to be mock code so probably not super important, but worth to keep in mind for future such changes.

Reply:
Oh, good to know!  Very subtle indeed!  Thanks